server.js

```diff
@@ -1,49 +1,43 @@
-let categories = require('./my-app-react/src/data/categories.json')
-let products = require('./my-app-react/src/data/products.json')
+const dataService = require('./dataservice.js');
+const fromEntries = require('object.fromentries');
 
-const express = require('express')
-const app = express()
+const express = require('express');
+const app = express();
 
 app.get('/info', (req, res) => {
-    info = [
-        {"serverName": "localhost5000"},
-        {"sampleName": "chinemerem-react"},
-        {"serverVersion": "1.0.0"}
-    ]
+    const serverInfo = require('package')
+    const info = [
+        {"serverName": serverInfo.name},
+        {"serverVersion": serverInfo.version}
+    ];
     res.json(info)
-})
+});
 
 app.get('/products/all', (req, res) => {
-    res.json(products)
-})
+    const productsMap = dataService.getCombinedProductMap();
+    res.send( fromEntries(productsMap));
```

Comment:
Why not just return `productsMap` ? 
```suggestion
    res.send(productsMap);
```
or 
```suggestion
    res.send(Object.fromEntries(productsMap));
```
The second one doesn't require any extra dependency.

Reply:
Object.fromEntries() is not a valid function in nodeJS. I've repeated this a lot of times.

Reply:
@felixfavour Then how come [it's documented nicely](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/fromEntries).

Check your compiler and make sure it complies with ES6.